### PR TITLE
applications: asset_tracker_v2: remove GPS driver dependency

### PR DESCRIPTION
**Pull request by tokangas**
Removed usage of the GPS driver API because the driver has been deprecated and will be removed.

Signed-off-by: Tommi Kangas <tommi.kangas@nordicsemi.no>

### CODE REVIEW

File: applications/asset_tracker_v2/src/cloud/cloud_codec/json_protocol_names.h [hunk 1]
```diff
@@ -81,6 +81,16 @@
 #define DATA_AGPS_REQUEST_TAC           "area"
 #define DATA_AGPS_REQUEST_TYPES         "types"
 
+#define DATA_AGPS_REQUEST_TYPE_UTC_PARAMETERS		 1
+#define DATA_AGPS_REQUEST_TYPE_EPHEMERIDES		 2
+#define DATA_AGPS_REQUEST_TYPE_ALMANAC			 3
+#define DATA_AGPS_REQUEST_TYPE_KLOBUCHAR_CORRECTION	 4
+#define DATA_AGPS_REQUEST_TYPE_NEQUICK_CORRECTION	 5
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/src/cloud/cloud_codec/json_protocol_names.h:88]:
```suggestion
#define DATA_AGPS_REQUEST_TYPE_NEQUICK_CORRECTION	 5
```
NEQUICK is not supported in the modem.

File: applications/asset_tracker_v2/src/cloud/cloud_codec/json_protocol_names.h [hunk 2]
```diff
@@ -166,6 +176,16 @@
 #define DATA_AGPS_REQUEST_TAC           "area"
 #define DATA_AGPS_REQUEST_TYPES         "types"
 
+#define DATA_AGPS_REQUEST_TYPE_UTC_PARAMETERS		 1
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/src/cloud/cloud_codec/json_protocol_names.h:179]: Duplicate defines.

**Review by coderbyheart — CHANGES_REQUESTED** [review]

File: applications/asset_tracker_v2/src/cloud/cloud_codec/json_protocol_names.h (hunk 1 above)
Comment (coderbyheart) [reply applications/asset_tracker_v2/src/cloud/cloud_codec/json_protocol_names.h:88]: ... so the application should not request it at all.

File: applications/asset_tracker_v2/src/cloud/cloud_codec/json_protocol_names.h (hunk 2 above)
Comment (simensrostad) [reply applications/asset_tracker_v2/src/cloud/cloud_codec/json_protocol_names.h:179]: A lot of these defines needs to be defined for both Azure, AWS and nRF Cloud builds so there might be duplicates. I don't see an issue with having only one set of defines for values that are identical across cloud backends. We could even have separate `_protocol_names.h` files for each cloud backend to make the abstraction clearer.

**Review by simensrostad — APPROVED** [review]

File: applications/asset_tracker_v2/src/cloud/cloud_codec/json_protocol_names.h (hunk 2 above)
Comment (tokangas) [reply applications/asset_tracker_v2/src/cloud/cloud_codec/json_protocol_names.h:179]: Yes, I just followed the convention that had been used earlier. There's a lot of stuff which is defined twice in the same way depending on CONFIG_NRF_CLOUD_MQTT.

Should I do something in this PR?

File: applications/asset_tracker_v2/src/cloud/cloud_codec/json_protocol_names.h (hunk 1 above)
Comment (tokangas) [reply applications/asset_tracker_v2/src/cloud/cloud_codec/json_protocol_names.h:88]: Support for NeQuick is (at least mostly) in place, although it's not currently used by the modem. It's even supported in the GNSS RPC API. Modem doesn't ask for it, so the application should not request it from the cloud either.

File: applications/asset_tracker_v2/src/cloud/cloud_codec/json_protocol_names.h (hunk 2 above)
Comment (simensrostad) [reply applications/asset_tracker_v2/src/cloud/cloud_codec/json_protocol_names.h:179]: Its not really in the scope of the PR so I wouldn't worry about it.

**Comment by tokangas** [comment]
Fixed json_common unit tests.

**Review by jtguggedal — APPROVED** [review]

File: applications/asset_tracker_v2/src/cloud/cloud_codec/json_protocol_names.h (hunk 1 above)
Comment (coderbyheart) [reply applications/asset_tracker_v2/src/cloud/cloud_codec/json_protocol_names.h:88]: I don't like have non-functional code in the codebase, but this is not a blocker.

**Review by coderbyheart — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**